cockroachdb/templates/statefulset.yaml

```
@@ -339,6 +342,9 @@ spec:
           secret:
             secretName: {{ template "cockroachdb.fullname" . }}-log-config
       {{- end }}
+      {{- with .Values.securityContext }}
+      securityContext: {{- toYaml . | nindent 8 }}
+      {{- end }}
```

Rather than passing in the raw security context, could we make this work like the `Values.storage.persistentVolume.enabled` flag? (Likewise for the StatefulSet).

We should also add a comment or a check for the CockroachDB version that advises people to disable it if running an older version of CockroachDB. @yecs1999 could you confirm the minimum versions we'd need to require?

```suggestion
      {{- if .Values.securityContext.enabled }}
      securityContext:
        fsGroup: 1000
        runAsGroup: 1000
        runAsUser: 1000
        runAsNonRoot: true
      {{- end }}
```

---

Good catch. @grohan2002 do you know if you can test this with the version tag [here](https://github.com/cockroachdb/helm-charts/blob/d7e28f049f381c76879d9394eb26123196682a56/cockroachdb/values.yaml#L2-L4) changed to 22.1.1 and 21.2.12? Both of these versions are below the minimum required. If there is an error there then we need to have some kind of switch so any versions that are unsupported (below 21.2.13 or below 22.1.2) will not have this securitycontext field populated.